src/network/stream_reader.rs

```diff
@@ -14,78 +14,53 @@
 
 //! Stream reader.
 //!
-//! This module defines the `StreamReader` struct and its implementation which
-//! is used for parsing an incoming stream into separate `RawNetworkMessage`s,
-//! handling assembling messages from multiple packets, or dealing with partial
-//! or multiple messages in the stream (e.g. when reading from a TCP socket).
+//! Deprecated
+//!
+//! This module defines `StreamReader` struct and its implementation which is used
+//! for parsing incoming stream into separate `Decodable`s, handling assembling
+//! messages from multiple packets or dealing with partial or multiple messages in the stream
+//! (like can happen with reading from TCP socket)
 //!
-
-use prelude::*;
 
 use core::fmt;
-use io::{self, Read};
+use io::Read;
 
 use consensus::{encode, Decodable};
 
 /// Struct used to configure stream reader function
 pub struct StreamReader<R: Read> {
     /// Stream to read from
     pub stream: R,
```

Comment:
Shouldn't we use `BufReader` to avoid performance issues in old code?

Reply:
Andrew thinks (https://github.com/rust-bitcoin/rust-bitcoin/pull/680#issuecomment-999914979) it's the caller's responsibility to wrap if needed.

I am not strongly for using it or not using it, but for sure the main issue this PR is solving and which is our responsibility is not attempting to decode 16 times the same 1MB block

Reply:
I mean `StreamReader` was originally intended to be buffered. So by not using `BufReader` internally we create (I think significant) performance regression for those who update library without changing the deprecated code. It might be less bad to just remove it if we don't use `BufReader` internally.

Reply:
> I mean StreamReader was originally intended to be buffered

ah that's true, let's see what @apoelstra thinks

Reply:
I mean, if we deprecate/remove it we might as well also change it to use BufReader internally :P